src/pointer-analysis/value_set.cpp

```diff
@@ -1203,9 +1204,9 @@ void value_sett::assign(
                 "type:\n"+type.pretty();
 
         rhs_member=make_member(rhs, name, ns);
-
-        assign(lhs_member, rhs_member, ns, is_simplified, add_to_sets);
       }
+
+      assign(lhs_member, rhs_member, ns, false, add_to_sets);
```

Comment:
If I'm reading this right the ID_unknown case used to be silently ignored? I suspect this increase in pessimism / improvement in soundness will have side effects, consider doing that as a separate PR since that seems orthogonal to simplifying unions?